keys/key_store_file.go

```diff
@@ -99,9 +100,15 @@ func IsValidKeyJson(j []byte) []byte {
 	return nil
 }
 
+func NewKeyStore(dir string, AllowBadFilePermissions bool, logger *logging.Logger) KeyStore {
+	return KeyStore{keysDirPath: dir, AllowBadFilePermissions: AllowBadFilePermissions, logger: logger.WithScope("KeyStore")}
```

Comment:
This would be applicable for a `Scope` too but needs `ComponentKey` primarily. So could we have:

```go
logger: logger.With(structure.ComponentKey, "keys").WithScope("NewKeyStore")
```

Also let's make this initialisation multiline